datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -1696,6 +1696,18 @@ mod tests {
             DataType::UInt64,
             vec![11u64, 6u64, 7u64],
         );
+        test_coercion!(
+            UInt16Array,
+            DataType::UInt16,
+            vec![3u16, 2u16, 3u16],
+            UInt64Array,
+            DataType::UInt64,
+            vec![10u64, 6u64, 5u64],
+            Operator::BitwiseOr,
+            UInt64Array,
+            DataType::UInt64,
+            vec![11u64, 6u64, 7u64],
+        );
```

Comment:
Hmm, there is already unsigned `test_coercion` test for `BitwiseOr`. For `BitwiseAnd` and `BitwiseXor`, they have both unsigned and signed `test_coercion` test, so I guess that you actually want to add signed `test_coercion` test?